Controller.php

```diff
@@ -107,6 +111,17 @@ private function checkDays()
         return $daysToCompute;
     }
 
+    private function checkCustomPiwikUrl()
+    {
+        $customPiwikUrl = Common::getRequestVar('customPiwikUrl', false, 'string');
+
+        if (!$customPiwikUrl = PageUrl::getUrlIfLookValid($customPiwikUrl)) {
```

Comment:
the method `PageUrl::getUrlIfLookValid` is not tagged with `@api` - ideally plugins will only use public apis method. here and in the other `checkCustomPiwikUrl` you can use the @api method: `UrlHelper::isLookLikeUrl` (I just added `@api` to this method in core)